app/uk/gov/hmrc/exports/services/mapping/declaration/BorderTransportMeansBuilder.scala

```diff
@@ -72,9 +67,11 @@ class BorderTransportMeansBuilder @Inject()(countriesService: CountriesService)
   }
 
   private def appendDepartureTransport(data: Transport, transportMeans: Declaration.BorderTransportMeans): Unit =
-    data.borderModeOfTransportCode.foreach { code =>
-      val modeCode = new BorderTransportMeansModeCodeType()
-      modeCode.setValue(code)
-      transportMeans.setModeCode(modeCode)
+    data.borderModeOfTransportCode.foreach { modeOfTransportCode =>
+      if (modeOfTransportCode.isDefined) {
```

Comment:
Should this not be testing for (and excluding) modeOfTransportCode.value="" (i.e. Empty)?

Reply:
Is it because a value of empty string in the payload is converted to None?  It must be that.

Reply:
`isDefined` is a method in `ModeOfTransportCode`. It basically checks if the object is not of instance ` ModeOfTransportCode.Empty`
Empty String in json will be converted to the `Empty` instance. That was my main question if this is correct.

Reply:
I missed the isDefined method.  I read it as modeOfTransportCode was an Option.  Makes sense now.